src/Sylius/Behat/Context/Ui/Admin/ManagingCurrenciesContext.php

```diff
@@ -0,0 +1,319 @@
+    public function iEnableIt()
+    {
+        $this->updatePage->enable();
+    }
+
+    /**
+     * @When I disable it
+     */
+    public function iDisableIt()
+    {
+        $this->updatePage->disable();
+    }
+
+    /**
+     * @When I save my changes
+     * @When I try to save my changes
+     */
+    public function iSaveMyChanges()
+    {
+        $this->updatePage->saveChanges();
+    }
+
+    /**
+     * @Then I should be notified about successful edition
+     */
+    public function iShouldBeNotifiedAboutSuccessfulEdition()
+    {
+        $this->notificationChecker->checkEditionNotification(self::RESOURCE_NAME);
+    }
+
+    /**
+     * @Then /^(this currency) should be disabled$/
+     */
+    public function thisCurrencyShouldBeDisabled(CurrencyInterface $currency)
+    {
+        $this->indexPage->open();
+
+        Assert::true(
+            $this->indexPage->isCurrencyDisabled($currency),
+            sprintf('Currency %s should be disabled but it is not.', $currency->getCode())
+        );
+    }
+
+    /**
+     * @Then /^(this currency) should be enabled$/
+     */
+    public function thisCurrencyShouldBeEnabled(CurrencyInterface $currency)
+    {
+        $this->indexPage->open();
+
+        Assert::true(
+            $this->indexPage->isCurrencyEnabled($currency),
+            sprintf('Currency %s should be enabled but it is not.', $currency->getCode())
+        );
+    }
+
+    /**
+     * @When I change exchange rate to :exchangeRate
+     */
+    public function iChangeExchangeRateTo($exchangeRate)
+    {
+       $this->updatePage->changeExchangeRate($exchangeRate);
+    }
+
+    /**
+     * @Then this currency should have exchange rate :exchangeRate
+     */
+    public function thisCurrencyShouldHaveExchangeRate($exchangeRate)
+    {
+        Assert::true(
+            $this->updatePage->assertExchangeRate($exchangeRate),
+            sprintf('Currency exchange rate should be equal %s, but is not.', $exchangeRate)
+        );
+    }
+
+    /**
+     * @Then the code filed should be disabled
+     */
+    public function theCodeFiledShouldBeDisabled()
+    {
+        Assert::true(
+            $this->updatePage->isCodeFieldDisabled(),
+            'Code field should be disabled but is not.'
+        );
+    }
+
+    /**
+     * @Then I should be notified that currency code must be unique
+     */
+    public function iShouldBeNotifiedThatCurrencyCodeMustBeUnique()
+    {
+        Assert::true(
+            $this->createPage->checkValidationMessageFor('code', 'Currency code must be unique.'),
+            'Unique code violation message should appear on page, but it does not.'
+        );
+    }
+
+    /**
+     * @Then there should still be only one currency with :element :code
+     */
+    public function thereShouldStillBeOnlyOneCurrencyWithCode($element, $codeValue)
+    {
+        $this->indexPage->open();
+
+        Assert::true(
+            $this->indexPage->isResourceOnPage([$element => $codeValue]),
```

Comment:
The method `isResourceOnPage` return true only if amount of rows with given parameters is equal to 1. So if there was more than one resource it would return false. 
I can create custom method for currency's indexPage, but it will be do the same thing. Maybe `isResourceOnPage` shold be renamed to `isOnlyOneResourceOnPage` or something like this?